examples/kfto-sft-feast-rag/sft_feast_rag_model.ipynb

```
@@ -0,0 +1,1292 @@
+    "    # Question Encoder\n",
+    "    question_encoder_model_name_or_path = \"facebook/dpr-question_encoder-single-nq-base\"\n",
+    "    question_encoder_tokenizer = DPRQuestionEncoderTokenizer.from_pretrained(question_encoder_model_name_or_path)\n",
+    "    question_encoder_model = DPRQuestionEncoder.from_pretrained(\n",
+    "        question_encoder_model_name_or_path,\n",
+    "        trust_remote_code=model_args.trust_remote_code,\n",
+    "        torch_dtype=model_args.torch_dtype\n",
+    "    )\n",
+    "\n",
+    "    # Generator Model to be fine-tuned\n",
+    "    generator_model_name_or_path = parameters[\"model_name_or_path\"]\n",
+    "    generator_tokenizer = AutoTokenizer.from_pretrained(\n",
```

Ruff (0.11.9)

Undefined name `parameters`

(F821)